dependency/repository/packagist.go

```diff
@@ -81,7 +81,7 @@ func NewPackagist(instance string, httpClient *http.Client) (*PackagistClient, e
 
 // GetPackageByName returns a package by a given name
 func (c *PackagistClient) GetPackageByName(name string) (*PackagistPackage, *http.Response, error) {
-	u := fmt.Sprintf("%s/packages%s.json", c.url.String(), filepath.Clean("/"+name))
+	u := fmt.Sprintf("%s/packages%s.json", c.url.String(), strings.Replace(filepath.Clean("/"+name), "\\", "/", -1))
```

Comment:
In https://github.com/andygrunwald/perseus/issues/6#issuecomment-700061241 you pointed out that the issue might be the usage of `filepath.Clean`. What do you think about removing the `filepath.Clean` reference and replacing it with an URL fitting pattern? e.g. typical URL encoding?
Have a look at https://www.urlencoder.io/golang/ to check if one of those might be a good choice.